clustered/client/src/main/java/org/ehcache/clustered/client/internal/store/SimpleClusterTierClientEntity.java

```diff
@@ -222,15 +236,43 @@ private EhcacheEntityResponse invokeInternal(TimeoutDuration timeLimit, EhcacheE
     }
   }
 
-  private InvokeFuture<EhcacheEntityResponse> internalInvokeAsync(EhcacheEntityMessage message, boolean track)
-        throws MessageCodecException {
-    return endpoint.beginInvoke().message(message).invoke();
+  private InvokeFuture<EhcacheEntityResponse> invokeInternal(InvocationBuilder<EhcacheEntityMessage, EhcacheEntityResponse> invocationBuilder, long timeout, EhcacheEntityMessage message, boolean track) throws TimeoutException {
+    boolean interrupted = Thread.interrupted();
+    try {
+      long end = System.nanoTime() + timeout;
```

Comment:
There is something weird with the timeout. I think it could be a endTime to simplify. Because if goes from timeout duration to end time to timeout duration.

Reply:
Done.